test/test_util.py

```diff
@@ -256,6 +258,39 @@ def test_make_headers(self):
             make_headers(disable_cache=True),
             {'cache-control': 'no-cache'})
 
+    def test_rewind_body(self):
+        body = io.BytesIO(b'test data')
+        self.assertEqual(body.read(), b'test data')
+
+        # Assert the file object has been consumed
+        self.assertEqual(body.read(), b'')
+
+        # Rewind it back to just be b'data'
+        rewind_body(body, 5)
+        self.assertEqual(body.read(), b'data')
+
+    def test_rewind_body_failed_tell(self):
+        body = io.BytesIO(b'test data')
+        body.read()  # Consume body
+
+        # Simulate failed tell()
+        self.assertRaises(UnrewindableBodyError, rewind_body, body, object())
+
+    def test_rewind_body_bad_position(self):
+        body = io.BytesIO(b'test data')
+        body.read()  # Consume body
+
+        # Pass non-integer position
+        self.assertRaises(UnrewindableBodyError, rewind_body, body, None)
```

Comment:
For invalid positions, should we throw `UnrewindableBodyError`, or simply a `ValueError`? No strong guidance here, just a question work thinking about.